pkg/monitoring/metrics/operator_metrics.go

```diff
@@ -119,30 +117,20 @@ func IsHCOMetricHyperConvergedExists() (bool, error) {
 	return value == hyperConvergedExists, nil
 }
 
-func SetHCOSystemHealthy() {
-	systemHealthStatus.Reset()
-	systemHealthStatus.WithLabelValues("healthy").Set(SystemHealthStatusHealthy)
-}
-
-func SetHCOSystemWarning(reason string) {
-	systemHealthStatus.Reset()
-	systemHealthStatus.WithLabelValues(reason).Set(SystemHealthStatusWarning)
-}
-
-func SetHCOSystemError(reason string) {
-	systemHealthStatus.Reset()
-	systemHealthStatus.WithLabelValues(reason).Set(SystemHealthStatusError)
+func SetHCOMetricSystemHealthStatus(status float64) {
+	systemHealthStatus.Set(status)
 }
 
 // GetHCOMetricSystemHealthStatus returns current value of gauge. If error is not nil then value is undefined
```

Comment:
@avlitman is the comment here correct?

Reply:
Yes it was before the changes to the metric

Reply:
which maybe problematic now that I think about it to return 0.. but this is how it was before any changes to the metric. WDYT?

Reply:
Before the code refactor it look like this:

```
// GetHCOMetricSystemHealthStatus returns current value of gauge. If error is not nil then value is undefined
func (hm *hcoMetrics) GetHCOMetricSystemHealthStatus() (float64, error) {
	return hm.GetMetricValue(HCOMetricSystemHealthStatus, nil)
}
```

so we can just drop the if maybe

Reply:
anyway this function is called only for testing purposes